### DISCUSSION TITLE
Create in-memory index ?

### DISCUSSION THREAD

**Discussion by bubbajoe** [apache/datafusion#6005] [Q&A] [answered]
Hello, this might be a stupid question but is it possible to create an in-memory index on a table?

Obviously the purpose would be to improve searches / ordering. Thanks

**Comment by jaylmiller** [comment, marked as answer]
You can implement your own `TableProvider` to do this. I don't believe this is built in to any of the `TableProvider`s that ship with DataFusion. Checkout https://github.com/apache/arrow-datafusion/blob/main/datafusion-examples/examples/custom_datasource.rs

**Reply by Jefffrey** [reply]
Also another discussion with in-depth discussion on indexing: https://github.com/apache/datafusion/discussions/9963
